spec.md

```diff
@@ -1616,23 +1617,34 @@ A register is initialized with an indeterminate value (see [@sec:indeterminate-v
 
 # Conditionals
 
-Conditional statements define one or more regions where the execution of operations (declarations, statements, and expressions) inside the regions is augmented by a *condition*.
-The evaluation behavior of operations is operation-dependent:
+Conditional statements define one or more regions where the operations (declarations, statements, and expressions) inside a region are *conditionally executed*.
+I.e., the operations within the region execute depending on the value of the condition and the kind of operation.
 
--   Connections (see [@sec:connections]) only drive their sink values when the product (logical and) of all conditions that are parents of the connect, but not parents of the sink declaration evaluates to true.
--   Commands (see [@sec:commands]) only evaluate when the product (logical and) of all parent conditions evaluates to true.
--   The evaluation of all other operations in conditional regions is unaffected by the condition.
-    E.g., register, wire, and node declarations as well as module instantiation statements and primitive operation expressions may occur in conditional regions but never evaluate conditionally.
+## Conditional Execution
+
+The *conditional execution* of operations in the region of a conditional statement is operation dependent:
+
+-   Connections (see [@sec:connections]) only execute when the product (logical and) of all conditions that are parents of the connect, but not parents of the sink declaration evaluates to true.
+-   Commands (see [@sec:commands]) only execute when the product (logical and) of all parent conditions evaluates to true.
```

Comment:
I'm not sure what you mean by "all" conditions, since `when` statements only have a single condition.

Your use of the term "parents" here suggests in my mind you're trying to define a new notion on the spot: "a **parent** (`when`/condition?) of a statement is any (`when`/condition of a `when` statement) which appears as the ancestor in the abstract syntax tree of the given statement." -- And even this is too cramped a definition.

Reply:
This is trying to define how the following works. It may conflate the notion of a parent region from a parent condition. The notion of regions and parents is pretty standard in LLVM and MLIR. (I'm not attempting to define something non-standard on the spot!):

``` firrtl
when a:
  wire x: UInt<1>
  when b:
    when c:
      connect x, y
```

This has three regions and associated conditions: one for a, one for b, and one for c. a is a parent of b and b is a parent of c. This is then trying to explain that the connection to `x` is conditioned on `b` and `c`, but not `a`. It may be more reasonable to not define it this way and instead say that the connection to `x` is _not_ conditional with respect to any parent conditions of `x`.

WDYT?

How would you describe this?

Reply:
I've added clarifications of what a "parent condition" is in https://github.com/chipsalliance/firrtl-spec/pull/215/commits/052f1ebf01f19f3efcedfcec229d5256af646a40.